### PR TITLE
Reading AuthenticationProperties from SignOutContext

### PR DESCRIPTION
**Pull request by Praburaj**
This will enable users to set a specific redirect uri and call signout.

This will depend on https://github.com/aspnet/HttpAbstractions/pull/218

@brentschmaltz @Tratcher @HaoK

### CODE REVIEW

**Comment by HaoK** [comment]
Change looks good, is there any way we can add a test verifying the scenario/what this is supposed to fix?

File: src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs [hunk 1]
```diff
@@ -77,7 +76,7 @@ protected override void ApplyResponseGrant()
                 // Set End_Session_Endpoint in order:
                 // 1. properties.Redirect
                 // 2. Options.Wreply
-                AuthenticationProperties properties = new AuthenticationProperties();
+                var properties = new AuthenticationProperties(signout.Properties);
                 if (properties != null && !string.IsNullOrEmpty(properties.RedirectUri))
```
Comment (PinpointTownes) [inline src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs]: You can remove the `properties` null check, it's obviously useless :smile:

Comment (Praburaj) [inline src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs]: You're right. I'll remove that.

**Comment by Praburaj** [comment]
@HaoK sure I can add a test. Let me update the PR.

**Comment by Praburaj** [comment]
@HaoK Added test.

File: test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs [hunk 2]
```diff
@@ -154,6 +154,22 @@ public class OpenIdConnectMiddlewareTests
         }
 
         [Fact]
+        public async Task SignOutWith_Specifi_RedirectUri_From_Authentication_Properites()
```
Comment (HaoK) [inline test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs]: nit: Specifi[c] typo

**Comment by HaoK** [comment]
looks good to me :shipit:

File: test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs (hunk 2 above)
Comment (Praburaj) [inline test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs]: Fixed.

**Comment by muratg** [comment]
:shipit:

File: test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs [hunk 3]
```diff
@@ -154,6 +154,22 @@ public class OpenIdConnectMiddlewareTests
...
+        {
+            var stateFormat = new PropertiesDataFormat(new EphemeralDataProtectionProvider().CreateProtector("OpenIdConnectTest"));
```
Comment (PinpointTownes) [inline test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs]: Unused variable?

Comment (Praburaj) [inline test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs]: Thanks. Removed this var from a bunch of tests.

**Comment by brentschmaltz** [comment]
This is a needed feature, and it helps. I think it could rate higher on the usability scale if users could set named properties like we do on the OpenIdConnectMessage.

**Comment by Praburaj** [comment]
Merged : https://github.com/aspnet/Security/commit/b7c8af8503831ba4becaa93b9453510b81c4d30e

**Pull request closed without merging**